packages/as-types/assembly/safeMath.ts

```diff
@@ -26,9 +26,7 @@ export class SafeMath {
    * @throws if the operation results in a number bigger than u64.MAX_VALUE.
    */
   static add(a: u64, b: u64): u64 {
-    assert(a + b >= a, 'SafeMath: addition overflow');
-    assert(a + b >= 0, 'SafeMath: addition overflow');
-    assert(a + b <= u64.MAX_VALUE, 'SafeMath: addition overflow');
+    assert(b < u64.MAX_VALUE - a, 'SafeMath: addition overflow');
```

Comment:
b = u64.MAX_VALUE and a = 0 shall pass
```suggestion
    assert(b <= u64.MAX_VALUE - a, 'SafeMath: addition overflow');
```